packages/kbn-unified-field-list/src/services/field_existing/field_existing_utils.ts

```diff
@@ -61,7 +61,7 @@ export async function fetchFieldExistence({
   const newFields = existingFieldList.filter((field) => !dataView.getFieldByName(field.name));
   // refresh the data view in case there are new fields
   if (newFields.length) {
-    await dataViewsService.refreshFields(dataView, false);
+    await dataViewsService.refreshFields(dataView, false, true);
```

Comment:
@kertal Do we need to wait here until it finishes?

Reply:
yes, else it wouldn't work in Lens (currently), which refreshes the field list, taking the most recent data view, if there are new fields. if we don't wait, we could run into the situation, that fields have not been loaded yet

Reply:
I need to double check this, because I think when we don't wait for the field result, and this request takes longer,  Lens is refreshing the field list using the previous completed data view fields version, while Discover is taking advantage of merging the old with the new fields ... I could be very wrong about his of course. If this wasn't the case, yes, we could think of removing the wait, to get faster results / rendering, while still refreshing the fields in the background

Reply:
I've checked by removing the `await` of the `refreshFields` function, and adding a 5s timeout inside the function to simulate a long running field refresh. In my test, the new field didn't show up. This is high likely because:
1. When there are new fields, the field list of Lens has to be refreshed
https://github.com/elastic/kibana/blob/d024027cec062a59ae8bbb0b6f15f2a7f00a9320/x-pack/plugins/lens/public/datasources/form_based/datapanel.tsx#L301
2. This triggers a reloading of the Lens-DataViews https://github.com/elastic/kibana/blob/d024027cec062a59ae8bbb0b6f15f2a7f00a9320/x-pack/plugins/lens/public/datasources/form_based/datapanel.tsx#L283
3. And since when this happens, the `refreshFields` function hasn't completed, new fields don't show up

... and I tested in Discover, where the field was added unmapped, for high likely another reason

Would it be possible to remove the await for refreshFields, in theory yes, but it would need definitely changes (something we can do further down the road)

Reply:
@kertal Thanks for looking into it! 

We could update the code where it separates Unmapped fields https://github.com/elastic/kibana/blob/main/packages/kbn-unified-field-list/src/hooks/use_grouped_fields.ts#L159-L161 to account for `newFields` in Discover. 

The issue is that making the UnifiedFieldList know about new fields is not enough, data view instance needs to know about them too as we call `dataView.getFieldByName` and `dataView.fields` in other places.  Calling ` await dataViewsService.refreshFields(dataView, false, true);` "fixes" the data view. But it also extends loading time.

Probably it's fine for now as our main objective is to automatically show new fields.

Reply:
I think we can optimize that, but since it's just refreshed when there are new fields it's not generally slowing the loading for other cases. So I don't think it's a priority (But a nice to have ... of course)